### PR TITLE
Show diagnostic nodes in Solution Explorer for CPS-based projects

### PR DESCRIPTION
**Pull request by tmeschter**
**Customer scenario**

image

In the old project system the user could see individual diagnostics listed under analyzers in the Solution Explorer. This change refactors and extends our code to provide this functionality in CPS-based projects as well.

**Bugs this fixes:**

https://github.com/dotnet/project-system/issues/88

**Workarounds, if any**

None.

**Risk**

Low.

**Performance impact**

Low. The feature is designed such that most of the work is done on-demand while the user expands the tree in Solution Explorer.

**Is this a regression from a previous update?**

This brings us close to parity with the experience we have in non-CPS-based projects.

**Root cause analysis:**

N/A

**How was the bug found?**

N/A

### CODE REVIEW

**Comment by tmeschter** [comment]
@dotnet/roslyn-analysis @dotnet/project-system Please review.

File: src/VisualStudio/Core/Test/SolutionExplorer/DiagnosticDescriptorComparerTests.vb [hunk 1]
```diff
@@ -28,7 +28,7 @@ Namespace Microsoft.VisualStudio.LanguageServices.UnitTests.SolutionExplorer
                 DiagnosticSeverity.Warning,
                 True)
 
-            Dim comparer = New DiagnosticItemSource.DiagnosticDescriptorComparer()
+            Dim comparer = New LegacyDiagnosticItemSource.DiagnosticDescriptorComparer()
```
Comment (srivatsn) [inline src/VisualStudio/Core/Test/SolutionExplorer/DiagnosticDescriptorComparerTests.vb]: Add tests for the CpsDiagnosticItem*?

**Comment by srivatsn** [comment]
LGTM modulo needing tests.

**Comment by tmeschter** [comment]
@srivatsn I've added a couple of basic unit tests for the `CpsDiagnosticItem` type. Beyond that we really need an integration test, which is being tracked by https://github.com/dotnet/project-system/issues/1607.

**Comment by tmeschter** [comment]
Tagging @MattGertz for Ask Mode approval.

**Comment by MattGertz** [comment]
Approved pending tests.

**Comment by tmeschter** [comment]
@dotnet-bot test windows_eta_open_prtest please

**Pull request merged**